protocol-units/da/m1/light-node-verifier/src/celestia/mod.rs

```diff
@@ -0,0 +1,193 @@
+use crate::{Error, Verified, VerifierOperations};
+use celestia_rpc::Client;
+use celestia_types::{nmt::Namespace, Blob};
+use m1_da_light_node_util::inner_blob::InnerBlob;
+use std::sync::Arc;
+
+#[derive(Clone)]
+pub struct Verifier {
+	/// The Celestia RPC client
+	pub client: Arc<Client>,
+	/// The namespace of the Celestia Blob
+	pub namespace: Namespace,
+}
+
+impl Verifier {
+	pub fn new(client: Arc<Client>, namespace: Namespace) -> Self {
+		Self { client, namespace }
+	}
+}
+
+#[tonic::async_trait]
+impl VerifierOperations<Blob, InnerBlob> for Verifier {
+	/// Verifies a Celestia Blob as a Valid InnerBlob
+	async fn verify(&self, blob: Blob, _height: u64) -> Result<Verified<InnerBlob>, Error> {
+		// Only assert that we can indeed get an InnerBlob from the Blob
+		let inner_blob = InnerBlob::try_from(blob).map_err(|e| Error::Internal(e.to_string()))?;
+
+		Ok(Verified::new(inner_blob))
+	}
+}
+
+pub mod pessimistic {
```

Comment:
Would be cleaner in a separate file. and what is `super::Verifier` now, "optimistic"? Maybe better to put it in a module alongside to keep the code balanced.

Reply:
I will move it out.